src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Spark.E2ETest.Utils;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.Sql.Streaming;
+using Xunit;
+
+namespace Microsoft.Spark.E2ETest.IpcTests
+{
+    [Collection("Spark E2E Tests")]
+    public class DataStreamReaderTests
+    {
+        private readonly SparkSession _spark;
+
+        public DataStreamReaderTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+        }
+
+        /// <summary>
+        /// Test signatures for APIs up to Spark 2.3.*.
+        /// </summary>
+        [Fact]
+        public void TestSignaturesV2_3_X()
+        {
+            DataStreamReader dsr = _spark.ReadStream();
+
+            Assert.IsType<DataStreamReader>(dsr.Format("parquet"));
+            Assert.IsType<DataStreamReader>(dsr.Schema("columnName bigint"));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", "value"));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", true));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", long.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", double.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Options(new Dictionary<string, string>()));
+
+            DataFrame data = _spark.Range(0, 5);
+            using (var tempDirectory = new TemporaryDirectory())
+            {
+                string filePath = Path.Combine(tempDirectory.Path, Guid.NewGuid().ToString());
+
+                data.Write().Json(filePath);
```

Comment:
can you check `DataFrameReaderTests.cs`? I believe we already have these files checked in such that you don't need to create these.

Reply:
Changed to use the reference files.